test/unit/math/opencl/copy_test.cpp

```diff
@@ -8,6 +8,209 @@
 #include <algorithm>
 #include <vector>
 
+TEST(MathMatrixGPU, copy_destroyed_constructor) {
```

Comment:
[Optional] Now that the bug is fixed we are not copying destroyed objects anymore. I would suggest changing the names of these tests from "copy_destroyed..." to "copy_rvalue..."

Reply:
Good call. Went and changed all `_destroyed_` to `_rvalue_` in copy_test.cpp. If that is fine with you, then these comments were all addressed.